spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/KStreamBinderConfiguration.java

```diff
@@ -16,36 +16,49 @@
 
 package org.springframework.cloud.stream.binder.kafka.streams;
 
-import org.apache.commons.logging.Log;
-import org.apache.commons.logging.LogFactory;
-
-import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanFactoryPostProcessor;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.kafka.KafkaAutoConfiguration;
 import org.springframework.boot.autoconfigure.kafka.KafkaProperties;
 import org.springframework.cloud.stream.binder.kafka.properties.KafkaBinderConfigurationProperties;
 import org.springframework.cloud.stream.binder.kafka.provisioning.KafkaTopicProvisioner;
 import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsBinderConfigurationProperties;
 import org.springframework.cloud.stream.binder.kafka.streams.properties.KafkaStreamsExtendedBindingProperties;
+import org.springframework.context.ApplicationContext;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
 
 /**
  * @author Marius Bogoevici
  * @author Gary Russell
  * @author Soby Chacko
  */
 @Configuration
+@Import({KafkaAutoConfiguration.class})
 public class KStreamBinderConfiguration {
 
-	private static final Log logger = LogFactory.getLog(KStreamBinderConfiguration.class);
-
-	@Autowired
-	private KafkaProperties kafkaProperties;
-
-	@Autowired
-	private KafkaStreamsExtendedBindingProperties kafkaStreamsExtendedBindingProperties;
+	@Bean
+	@ConditionalOnBean(name = "outerContext")
+	public BeanFactoryPostProcessor outerContextBeanFactoryPostProcessor() {
+		return beanFactory -> {
+			ApplicationContext outerContext = (ApplicationContext) beanFactory.getBean("outerContext");
+			beanFactory.registerSingleton("kafkaStreamsBinderConfigurationProperties", outerContext
+					.getBean(KafkaStreamsBinderConfigurationProperties.class));
+			beanFactory.registerSingleton("kafkaStreamsMessageConversionDelegate", outerContext
+					.getBean(KafkaStreamsMessageConversionDelegate.class));
+			beanFactory.registerSingleton("kafkaStreamsBindingInformationCatalogue", outerContext
+					.getBean(KafkaStreamsBindingInformationCatalogue.class));
+			beanFactory.registerSingleton("keyValueSerdeResolver", outerContext
+					.getBean(KeyValueSerdeResolver.class));
+			beanFactory.registerSingleton("kafkaStreamsExtendedBindingProperties", outerContext
```

Comment:
Soby, should it be something like
```
beanFactory.registerSingleton(KeyValueSerdeResolver.class.getSimpleName(), outerContext.getBean(KeyValueSerdeResolver.class));
```
We really don't care about the name to begin with, so why not just delegate it to class name?

Reply:
Done. Changed all those calls with that.